lib/sqlbuilder/batch.go

```diff
@@ -19,51 +14,35 @@ func newBatchInserter(inserter *inserter, size int) *BatchInserter {
 	b := &BatchInserter{
 		inserter: inserter,
 		size:     size,
-		next:     make(chan Inserter),
+		Values:   make(chan interface{}, size),
 	}
-	b.reset()
 	return b
 }
 
-func (b *BatchInserter) reset() {
-	b.values = make([][]interface{}, 0, b.size)
-}
-
-func (b *BatchInserter) flush() {
-	if len(b.values) > 0 {
-		clone := b.inserter.clone()
-		for i := range b.values {
-			clone.Values(b.values[i]...)
+func (b *BatchInserter) Next(dst interface{}) bool {
+	clone := b.inserter.clone()
+	i := 0
+	for value := range b.Values {
+		i++
+		clone.Values(value)
```

Comment:
It is valid, but the usual API encourages grouping values, like:

```go
sess.InsertInto("foo").Columns("bar", "baz").Value(1, 2).Value(3, 4).Value(5, 6)
```

How would you push a group like `(1, 2)` here?